app/models/user.rb

```diff
@@ -160,6 +160,7 @@ class User < ApplicationRecord
   validate :validate_mastodon_url
   validate :can_send_confirmation_email
   validate :update_rate_limit
+  validate :password_matches_confirmation, if: :encrypted_password_changed?
```

Comment:
This is an implementation detail of how Devise works, we have to check for `encrypted_password_changed?` not `password_changed?`:

```ruby
user.password = 'muchsecure'
user.password_changed?
# NoMethodError: undefined method `password_changed?' for #<User:0x000000011e7bf100>
user.encrypted_password_changed?
#=> true
```

Reply:
can you add a comment there to explain this?